src/clients/firewall.rb

```diff
@@ -57,11 +57,34 @@ def main
         SuSEFirewallCMDLine.Run 
         # GUI or TextUI
       else
-        # installation has other sequence
-        if Mode.installation
-          @ret = FirewallInstallationSequence()
+        # If FirewallD then use it's UI
+        if SuSEFirewall.is_a?(Yast::SuSEFirewalldClass)
+          # We can't do ncurces. Lets see if the firewalld-config
+          # is installed
+          Yast.import "PackageSystem"
+          Yast.import "UI"
+          if UI.TextMode()
+            UI.OpenDialog(
+              VBox(
+                Label(_("Your display can't support the 'firewall-config' UI.")),
+                Label(_("Either use the Yast2 command line or the 'firewall-cmd' utility.")),
+                PushButton( Opt(:default), "OK" ),
+              )
+            )
+            UI::UserInput();
+            UI::CloseDialog();
```

Comment:
Not needed, you can use already existing popup class - https://github.com/yast/yast-yast2/blob/8eb9214265fdf8afcae1531c5a0309799028d79d/library/general/src/modules/Popup.rb#L1439

so it can look like
```
if UI.TextMode()
  Yast::Popup.Notify(_("Your display can't support the 'firewall-config' UI.") + "<br>" + _("Either use the Yast2 command line or the 'firewall-cmd' utility.") )
  return false
end
```

or maybe warning or error can be option for you.